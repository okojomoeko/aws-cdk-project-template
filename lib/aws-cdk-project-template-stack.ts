import { Duration, Stack, StackProps } from 'aws-cdk-lib';
import * as sns from 'aws-cdk-lib/aws-sns';
import * as subs from 'aws-cdk-lib/aws-sns-subscriptions';
import * as sqs from 'aws-cdk-lib/aws-sqs';
import { Construct } from 'constructs';

export class AwsCdkProjectTemplateStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const queue = new sqs.Queue(this, 'AwsCdkProjectTemplateQueue', {
      visibilityTimeout: Duration.seconds(300)
    });

    const topic = new sns.Topic(this, 'AwsCdkProjectTemplateTopic');

    topic.addSubscription(new subs.SqsSubscription(queue));
  }
}
