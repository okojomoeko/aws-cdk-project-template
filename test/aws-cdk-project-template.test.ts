import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as AwsCdkProjectTemplate from '../lib/aws-cdk-project-template-stack';

test('SQS Queue and SNS Topic Created', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new AwsCdkProjectTemplate.AwsCdkProjectTemplateStack(
    app,
    'MyTestStack'
  );
  // THEN

  const template = Template.fromStack(stack);

  template.hasResourceProperties('AWS::SQS::Queue', {
    VisibilityTimeout: 300,
  });
  template.resourceCountIs('AWS::SNS::Topic', 1);
  expect(template.toJSON()).toMatchSnapshot();
});
