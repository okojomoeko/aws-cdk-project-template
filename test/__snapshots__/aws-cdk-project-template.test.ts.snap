// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SQS Queue and SNS Topic Created 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AwsCdkProjectTemplateQueue96EC2A4C": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "VisibilityTimeout": 300,
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "AwsCdkProjectTemplateQueueMyTestStackAwsCdkProjectTemplateTopicFED9FDCC8B1D3C9F": {
      "DependsOn": [
        "AwsCdkProjectTemplateQueuePolicyB1E310F7",
      ],
      "Properties": {
        "Endpoint": {
          "Fn::GetAtt": [
            "AwsCdkProjectTemplateQueue96EC2A4C",
            "Arn",
          ],
        },
        "Protocol": "sqs",
        "TopicArn": {
          "Ref": "AwsCdkProjectTemplateTopic340E5805",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "AwsCdkProjectTemplateQueuePolicyB1E310F7": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "sqs:SendMessage",
              "Condition": {
                "ArnEquals": {
                  "aws:SourceArn": {
                    "Ref": "AwsCdkProjectTemplateTopic340E5805",
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "sns.amazonaws.com",
              },
              "Resource": {
                "Fn::GetAtt": [
                  "AwsCdkProjectTemplateQueue96EC2A4C",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Queues": [
          {
            "Ref": "AwsCdkProjectTemplateQueue96EC2A4C",
          },
        ],
      },
      "Type": "AWS::SQS::QueuePolicy",
    },
    "AwsCdkProjectTemplateTopic340E5805": {
      "Type": "AWS::SNS::Topic",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
